### PR TITLE
[Doc] Chinese User Guide chapter1

### PR DESCRIPTION
**Pull request by zhjwy9343**
## Description
Add Chinese User Guide chapter 1

## Checklist
Please feel free to remove inapplicable items for your PR.
- [X] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [X] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

**Comment by jermainewang** [comment]
Please revert changes on `third_party/dmlc-core`.

File: docs/source/guide_cn/index.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+用户指南
```
Comment (mufeili) [inline docs/source/guide_cn/index.rst:1]: Remove this file for this PR.

File: docs/source/guide_cn/graph-basic.rst [hunk 2]
```diff
@@ -0,0 +1,23 @@
+.. _guide_cn-graph-basic:
+
+1.1 关于图的基本概念
+-----------------
+
+:ref:`(English Version) <guide-graph-basic>`
+
+图是用以保存实体及其关系的的结构，记为 :math:`G=(V, E)` 。图由两个集合组成，一是节点的集合 :math:`V` ，一个是边的集合 :math:`E` 。
```
Comment (mufeili) [inline docs/source/guide_cn/graph-basic.rst:8]: "图是用以保存实体及其关系的的结构" -> "图是用以表示实体及其关系的结构"

File: docs/source/guide_cn/graph-basic.rst [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+在边集 :math:`E` 中，一条边 :math:`(u, v) \in E` 连接一对节点 :math:`u` 和 :math:`v` ，表明两节点间存在关系。关系可以是无向的，
```
Comment (mufeili) [inline docs/source/guide_cn/graph-basic.rst:9]: 1. 已经说了"在边集 :math:`E` 中"，可以把:math:`(u, v) \in E`改为:math:`(u, v)`

File: docs/source/guide_cn/graph-basic.rst [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+如描述节点之间的对称关系；也可以是有向的，如描述非对称关系。例如，若用图对社交网络中人们的友谊关系进行建模，因为友谊是相互的，则边是无向的；
+若用图对Twitter用户的关注行为进行建模，则边是有向的。图可以是*有向的*或*无向的*，这取决于图中边的方向性。
```
Comment (mufeili) [inline docs/source/guide_cn/graph-basic.rst:11]: "图可以是*有向的*或*无向的*" -> "图可以是 *有向的* 或 *无向的*"，否则渲染有问题。

File: docs/source/guide_cn/graph-graphs-nodes-edges.rst [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+1.2 图、节点和边
+--------------
+
+:ref:`(English Version)<guide-graph-graphs-nodes-edges>`
+
+DGL使用一个唯一的整数来表示一个节点，称为点ID；并用对应的两个端点ID表示一条边。同时，DGL也会根据边被添加的顺序，
+给每条边分配一个唯一的整数编号，称为边ID。节点和边的ID都是从0开始构建的。在DGL的图里，所有的边都是有方向的，
+即边 :math:`(u, v)` 表示它是从节点 :math:`u` 指向节点 :math:`v` 的。
+
+对于多个节点，DGL使用一个一维的张量（如，PyTorch的Tensor类，TensorFlow的Tensor类或MXNet的ndarray类）来保存图的点ID，
```
Comment (mufeili) [inline docs/source/guide_cn/graph-graphs-nodes-edges.rst:12]: “一维的张量” -> “一维的整形张量”

File: docs/source/guide_cn/graph-graphs-nodes-edges.rst [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    >>> print(g.edges())
+    (tensor([0, 0, 0, 1]), tensor([1, 2, 3, 3]))
+    >>> # 获取边的对应端点和边ID
+    >>> print(g.edges(form='all'))
+    (tensor([0, 0, 0, 1]), tensor([1, 2, 3, 3]), tensor([0, 1, 2, 3]))
+
+    >>> # 如果具有最大ID的节点没有边，在创建图的时候，用户需要明确地指明节点的数量。
+    >>> g = dgl.graph((u, v), num_nodes=8)
+
+对于无向的图，用户需要为每条边都创建两个方向的边。可以 :func:`dgl.to_bidirected` 函数使用来实现这个目的。
```
Comment (mufeili) [inline docs/source/guide_cn/graph-graphs-nodes-edges.rst:53]: “可以 :func:`dgl.to_bidirected` 函数使用” -> “可以使用 :func:`dgl.to_bidirected` 函数”

**Comment by mufeili** [comment]
Can you also change L48 in `docs/source/guide/graph-feature.rst` from `  node or edge (e.g., see lines 10-11 in the above example).` to `  node or edge (e.g., see lines 16-18 in the above example).`？

File: docs/source/guide_cn/graph-external.rst [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+许多知名Python库(如Pandas)可以将该类型数据加载到python对象(如 :class:`numpy.ndarray`)中，
+进而使用这些对象来构建DGLGraph对象。如果后端框架也提供了从磁盘中保存或加载张量的工具(如 :func:`torch.save`,  :func:`torch.load` )，
+可以遵循相同的原理来构建图。
+
+另见： `从成对的边 CSV 文件中加载 Karate Club Network 的教程 <https://github.com/dglai/WWW20-Hands-on-Tutorial/blob/master/basic_tasks/1_load_data.ipynb>`_。
+
+JSON/GML 格式
+""""""""""""
+
+如果对速度不太关注的话，读者可以参考NetworkX提供的工具来解析 `各种数据格式 <https://networkx.github.io/documentation/stable/reference/readwrite/index.html>`_，
```
Comment (mufeili) [inline docs/source/guide_cn/graph-external.rst:100]: “参考NetworkX” -> “使用NetworkX”

File: docs/source/guide_cn/graph-external.rst [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+DGL可以间接地从这些来源创建图。
+
+DGL 二进制格式
+""""""""""""
+
+DGL提供了API以从磁盘中保存和加载二进制格式的图。除了图结构，API也能处理特征数据和图级别的标签数据。
```
Comment (mufeili) [inline docs/source/guide_cn/graph-external.rst:106]: “从磁盘中保存和加载”->“从磁盘中加载或向磁盘保存”

File: docs/source/guide_cn/graph-external.rst [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+DGL也支持直接使用S3或HDFS的检查点图。参考手册提供了该用法的更多细节。
```
Comment (mufeili) [inline docs/source/guide_cn/graph-external.rst:107]: “直接使用S3或HDFS的检查点图” -> "从S3/HDFS中加载或向S3/HDFS保存图"

File: docs/source/guide_cn/graph-heterogeneous.rst [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+.. figure:: https://data.dgl.ai/asset/image/user_guide_graphch_2.png
+
+    一个异构图示例。该图具有两种类型的节点("用户"和"游戏")和两种类型的边("关注"和"玩")。
+
+创建异构图
+^^^^^^^^
+
+在DGL中，一个异构图由一系列子图构成，一个子图对应一种关系。每个关系由一个字符串三元组
+定义 ``(源节点类型, 边类型, 目标节点类型)`` 。由于这里的关系定义消除了边类型的歧义，DGL称
+它们为规范边类型。
```
Comment (mufeili) [inline docs/source/guide_cn/graph-heterogeneous.rst:20]: “它”前有一个额外的空格。

**Comment by mufeili** [comment]
Can you also remove

```
.. code::

    {relation1 : node_tensor_tuple1,
     relation2 : node_tensor_tuple2,
     ...}
```

in `docs/source/guide/graph-heterogeneous.rst`, starting from L23.

File: docs/source/guide_cn/graph-heterogeneous.rst [hunk 11]
```diff
@@ -0,0 +1,264 @@
...
+
+.. code::
+
+    >>> # 一个同构图
+    >>> dgl.heterograph({('node_type', 'edge_type', 'node_type'): (u, v)})
+    >>> # 一个二分图
+    >>> dgl.heterograph({('source_type', 'edge_type', 'destination_type'): (u, v)})
+
+与异构图相关联的 *metagraph* 就是图的模式。它指定节点集和节点之间的边的类型约束。*metagraph* 中的一个节点 :math:`u` 对应于
+相关异构图中的一个节点类型。*metagraph* 中的边 :math:`(u,v)` 表示在相关异构图中存在从 :math:`u` 型节点到 :math:`v` 型节点的边。
```
Comment (mufeili) [inline docs/source/guide_cn/graph-heterogeneous.rst:55]: “相关异构图中的一个节点类型”前多了一个空格。

File: docs/source/guide_cn/graph-heterogeneous.rst [hunk 12]
```diff
@@ -0,0 +1,264 @@
...
+.. note::
+
+    当边类型唯一地确定了源节点和目标节点的类型时，用户可以只使用一个字符串而不是字符串三元组来指定边类型。例如，
+    对于具有两个关系 ``('user', 'plays', 'game')`` 和  ``('user', 'likes', 'game')`` 的异构图，
+    只使用 ``'plays'`` 或 ``'like'`` 来指代这两个关系是可以的。
+
+从磁盘加载异构图
+^^^^^^^^^^^^^
+
+（CSV）格式
```
Comment (mufeili) [inline docs/source/guide_cn/graph-heterogeneous.rst:129]: “（CSV）格式” -> “逗号分隔值（CSV）”，与1.4保持一致。

File: docs/source/guide_cn/graph-heterogeneous.rst [hunk 13]
```diff
@@ -0,0 +1,264 @@
...
+""""""""""
+
+一种存储异构图的常见方法是在不同的CSV文件中存储不同类型的节点和边。下面是一个例子。
+
+.. code::
+
+    # data folder
```
Comment (mufeili) [inline docs/source/guide_cn/graph-heterogeneous.rst:136]: “data folder” -> “数据文件夹”

File: docs/source/guide_cn/graph-heterogeneous.rst [hunk 14]
```diff
@@ -0,0 +1,264 @@
...
+    data/
+    |-- drug.csv        # drug节点
+    |-- gene.csv        # gene节点
+    |-- disease.csv     # disease节点
+    |-- drug-interact-drug.csv  # drug-drug交互边
+    |-- drug-interact-gene.csv  # drug-gene交互边
```
Comment (mufeili) [inline docs/source/guide_cn/graph-heterogeneous.rst:142]: “交互” -> “相互作用”

File: docs/source/guide_cn/graph-heterogeneous.rst [hunk 15]
```diff
@@ -0,0 +1,264 @@
...
+    >>> g.edges['interacts'].data['he'] = th.zeros(2, 1)
+    >>> g.edges['treats'].data['he'] = th.zeros(1, 2)
+
+    >>> # 默认情况下不进行特征合并
+    >>> hg = dgl.to_homogeneous(g)
+    >>> 'hv' in hg.ndata
+    False
+
+    >>> # 拷贝边的特征
+    >>> # 对于要拷贝的特征，DGL假定不同类型的节点或边的合并特征具有相同的大小和数据类型
```
Comment (mufeili) [inline docs/source/guide_cn/graph-heterogeneous.rst:212]: “的合并特征” -> “要合并的特征”

**Comment by mufeili** [comment]
Can you also remove the blank space at the beginning of L248 in `docs/source/guide/graph-heterogeneous.rst`?

File: docs/source/guide_cn/graph-basic.rst (hunk 2 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/graph-basic.rst:8]: changed

File: docs/source/guide_cn/graph-basic.rst (hunk 4 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/graph-basic.rst:11]: changed

File: docs/source/guide_cn/graph-graphs-nodes-edges.rst (hunk 5 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/graph-graphs-nodes-edges.rst:12]: changed

File: docs/source/guide_cn/graph-graphs-nodes-edges.rst (hunk 6 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/graph-graphs-nodes-edges.rst:53]: changed

File: docs/source/guide_cn/graph-external.rst (hunk 7 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/graph-external.rst:100]: changed

File: docs/source/guide_cn/graph-external.rst (hunk 8 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/graph-external.rst:106]: changed

File: docs/source/guide_cn/graph-external.rst (hunk 9 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/graph-external.rst:107]: changed

File: docs/source/guide_cn/graph-heterogeneous.rst (hunk 10 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/graph-heterogeneous.rst:20]: changed

File: docs/source/guide_cn/graph-heterogeneous.rst (hunk 11 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/graph-heterogeneous.rst:55]: changed

File: docs/source/guide_cn/graph-heterogeneous.rst (hunk 12 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/graph-heterogeneous.rst:129]: changed

File: docs/source/guide_cn/graph-heterogeneous.rst (hunk 13 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/graph-heterogeneous.rst:136]: changed

File: docs/source/guide_cn/graph-heterogeneous.rst (hunk 14 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/graph-heterogeneous.rst:142]: changed

File: docs/source/guide_cn/graph-heterogeneous.rst (hunk 15 above)
Comment (zhjwy9343) [reply docs/source/guide_cn/graph-heterogeneous.rst:212]: change to "的需要合并的特征"

**Review by mufeili — APPROVED** [review]
LGTM.

**Pull request merged**